Library/Homebrew/dependencies_helpers.rb

```diff
@@ -49,6 +59,12 @@ def recursive_includes(klass, root_dependent, includes, ignores)
         keep ||= dep.test? && includes.include?("test?") && dependent == root_dependent
         keep ||= dep.build? && includes.include?("build?")
         klass.prune unless keep
+
+        next unless skip_recursive_build_dependents
+        next unless dep.build?
+        next if dependent == root_dependent && used_formulae.include?(dep.to_formula)
```

Comment:
I'm a bit bothered by the asymmetry between this condition and the similar one for test dependencies above. I think `--include-test --recursive` should be doing the same thing as `--include-build --recursive --skip-recursive-build-dependents`, except the former does it for test dependencies and the latter for build dependencies.

Except I keep getting wrong results if I drop the `used_formulae.include?` from here.

Reply:
Ok, I understand now. They're definitely not the same.